ckb/src/main/java/org/nervos/ckb/transaction/CkbTransactionBuilder.java

```diff
@@ -122,15 +182,28 @@ public TransactionWithScriptGroups build(Object... contexts) {
       }
 
       inputsCapacity += input.output.capacity;
-      // check if there is enough capacity for output capacity and change
-      long fee = calculateTxFee(tx, configuration.getFeeRate());
-      long changeCapacity = inputsCapacity - outputsCapacity - fee + reward;
-      CellOutput changeOutput = tx.outputs.get(changeOutputIndex);
-      byte[] changeOutputData = tx.outputsData.get(changeOutputIndex);
-      if (changeCapacity >= changeOutput.occupiedCapacity(changeOutputData)) {
-        tx.outputs.get(changeOutputIndex).capacity = changeCapacity;
-        enoughCapacity = true;
-        break;
+      final Long forceSmallChangeAsFee = getConfiguration().getForceSmallChangeAsFee();
+      if (forceSmallChangeAsFee != null) {
+        long fee = calculateTxFee(tx, configuration.getFeeRate());
+        long changeCapacity = inputsCapacity - outputsCapacity - fee + reward;
+        if (changeCapacity > 0 && changeCapacity < forceSmallChangeAsFee) {
```

Comment:
is it better to use `<=` ?
```suggestion
        if (changeCapacity > 0 && changeCapacity <= forceSmallChangeAsFee) {
```